tests/test_services.py

```diff
@@ -55,3 +55,126 @@ def __init__(self, min_dayobs, max_dayobs):
     assert result[0]["moon_illumination"] == 0.75
     assert "twilight_evening" in result[0]
     assert "twilight_morning" in result[0]
+
+
+def test_get_expected_exposures_normal_behaviour(monkeypatch):
+    """Test normal behavior: 3 days, each returning 100 visits."""
+
+    def fake_fetch(dayObs):
+        return {"nominal_visits": 100}
+
+    monkeypatch.setattr(
+        scheduler_service,
+        "fetch_sim_stats_for_night",
+        fake_fetch,
+    )
+
+    # 20240101–20240103 = 3 nights
+    result = scheduler_service.get_expected_exposures(20240101, 20240103)
+    assert result["sum"] == 300
+
+
+def test_get_expected_exposures_missing_nominal_visits(monkeypatch):
+    """
+    If the external call returns a dict without nominal_visits,
+    treat as zero.
+    """
+
+    def fake_fetch(dayObs):
+        return {}  # missing key
+
+    monkeypatch.setattr(
+        scheduler_service,
+        "fetch_sim_stats_for_night",
+        fake_fetch,
+    )
+
+    result = scheduler_service.get_expected_exposures(20240101, 20240101)
+    assert result["sum"] == 0
+
+
+def test_get_expected_exposures_inner_exception(monkeypatch):
+    """If one day raises inside loop, it should append 0 for that night."""
+
+    def fake_fetch(dayObs):
+        raise RuntimeError("boom")
+
+    monkeypatch.setattr(
+        scheduler_service,
+        "fetch_sim_stats_for_night",
+        fake_fetch,
+    )
+
+    # Two days, both should become zero
+    result = scheduler_service.get_expected_exposures(20240101, 20240102)
+    assert result["sum"] == 0
+
+
+def test_get_expected_exposures_partial_failures(monkeypatch):
+    """
+    Some days succeed, some fail.
+    Ensure mixed results handled correctly.
+    """
+
+    def fake_fetch(dayObs):
+        if dayObs == 20240101:
+            return {"nominal_visits": 50}
+        else:
+            raise Exception("fail")
+
+    monkeypatch.setattr(
+        scheduler_service,
+        "fetch_sim_stats_for_night",
+        fake_fetch,
+    )
+
+    result = scheduler_service.get_expected_exposures(20240101, 20240102)
+    # 50 + 0
+    assert result["sum"] == 50
+
+
+def test_get_expected_exposures_start_greater_than_end(monkeypatch):
+    """If start > end, loop never runs → sum = 0."""
+    # Should not need to call fetch at all,
+    # but monkeypatch anyway to avoid accidental calls
+    monkeypatch.setattr(
+        "rubin_sim.sim_archive.fetch_sim_stats_for_night",
+        lambda _: {"nominal_visits": 9999},
+    )
+
+    result = scheduler_service.get_expected_exposures(20240102, 20240101)
+    assert result["sum"] == 0
```

Comment:
Nice! I suggest to refactor this test to make it fail when/if `rubin_sim.sim_archive.fetch_sim_stats_for_night` is called. That way it will be easier to detect unexpected accidental calls.